### ISSUE TITLE
[hexagon] benign register transfers cause spurious warnings

### ISSUE THREAD

**Issue by androm3da** [llvm/llvm-project#60606]
```console
$ echo 'r0=#0x80000000' | ./bin/llvm-mc -arch=hexagon
	.text
<stdin>:1:1: warning: Signed/Unsigned mismatch
r0=#0x80000000
^
	{
		r0 = ##-2147483648
	}
```
